[sw] Invalidate icache in SRAM GDB test

dmcardle opened the pull request:
This commit also generalizes the test's success pattern so it does not care about the specific line number.

Fixes #15595

Signed-off-by: Dan McArdle <dmcardle@google.com>

alphan commented on sw/device/examples/sram_program/sram_program.c:
```diff
@@ -47,3 +46,10 @@ void sram_main() {
+// Reference functions that the debugger may wish to call. This prevents their
+// includes from appearing unused. There is probably no danger of the linker
+// removing unused functions thanks to our custom linker script.
+void debugger_support_functions() {
+  (void)icache_invalidate;
+}
```
Yeah, good point, this is needed because we use `--gc-sections` :)

alphan commented:
grumpy clang-format:
```
--- sw/device/examples/sram_program/sram_program.c	2022-10-20 16:26:01.000000000 +0000
+++ /dev/fd/63	2022-10-20 16:30:12.796100097 +0000
@@ -50,6 +50,4 @@
 // Reference functions that the debugger may wish to call. This prevents their
 // includes from appearing unused. There is probably no danger of the linker
 // removing unused functions thanks to our custom linker script.
-void debugger_support_functions() {
-  (void)icache_invalidate;
-}
+void debugger_support_functions() { (void)icache_invalidate; }
```

dmcardle commented:
Thanks, Alphan!

Fixed the formatting issue and updated the comment about dropping unused functions.

So, despite this being technically unnecessary per https://github.com/lowRISC/opentitan/issues/15595#issuecomment-1285838599, I still think it's a good idea because (a) the behavior of OpenOCD I described might be an implementation detail subject to change and (b) it draws attention to the need to clear the icache in case anyone models another test off of this.

alphan commented:
Merged since this change shouldn't affect the private CI.
